nixos/modules/programs/gpu-screen-recorder.nix

```diff
@@ -34,6 +40,66 @@ in {
       capabilities = "cap_sys_nice+ep";
       source = "${package}/bin/gpu-screen-recorder";
     };
+
+    /*
+      Tell nvidia driver to preserve video memory on suspend.
+      This is a workaround for an nvidia driver bug that breaks CUDA
+      (and gpu screen recorder) on suspend
+    */
+    environment.etc = {
+      "modprobe.d/gsr-nvidia.conf" = {
+        text = "options nvidia NVreg_PreserveVideoMemoryAllocations=1";
+      };
+    };
+
+    systemd.services.gpu-screen-recorder = {
+      description = "GPU Screen Recorder Service";
+      after = [ "network.target" ];
+      serviceConfig = {
+        EnvironmentFile = "-%h/.config/gpu-screen-recorder.env";
+        Environment = [
+          "WINDOW=screen"
+          "CONTAINER=mp4"
+          "QUALITY=very_high"
+          "CODEC=auto"
+          "AUDIO_CODEC=opus"
+          "AUDIO_DEVICE="
+          "SECONDARY_AUDIO_DEVICE="
+          "FRAMERATE=60"
+          "REPLAYDURATION=30"
+          "OUTPUTDIR=%h/Videos"
+          "MAKEFOLDERS=no"
+          "COLOR_RANGE=limited"
+          "KEYINT=2"
+          "ENCODER=gpu"
+          "RESTORE_PORTAL_SESSION=yes"
+        ];
+        ExecStart = ''
+          gpu-screen-recorder \
+            -v no \
+            -w "$WINDOW" \
+            -c "$CONTAINER" \
+            -q "$QUALITY" \
+            -k "$CODEC" \
+            -ac "$AUDIO_CODEC" \
+            -a "$AUDIO" \
+            -a "''${AUDIO_DEVICE:-$(${lib.getExe' pkgs.pulseaudio "pactl"} get-default-sink).monitor}" \
```

Comment:
pactl is no longer needed to use the default audio output device, the string "default_output" can now be used (this has also been fixed in the upstream systemd service file).
If you set AUDIO_DEVICE default value to default_output on line 66 then this line can be changed to -a "$AUDIO_DEVICE".
The above -a "$AUDIO" also doesn't do anything in this case so it can be removed.

Reply:
Thanks for the review!